hmrc-mongo-play-26/src/main/scala/uk/gov/hmrc/mongo/play/json/PlayMongoCollection.scala

```diff
@@ -32,28 +32,41 @@ class PlayMongoCollection[A: ClassTag](
   val collectionName: String,
   domainFormat: Format[A],
   optRegistry: Option[CodecRegistry] = None,
-  val indexes: Seq[IndexModel]
-)(implicit ec: ExecutionContext) extends MongoDatabaseCollection {
+  val indexes: Seq[IndexModel],
+  rebuildIndexes: Boolean = false
+)(implicit ec: ExecutionContext)
+    extends MongoDatabaseCollection {
 
   private val logger = Logger(getClass)
 
   val collection: MongoCollection[A] =
     CollectionFactory.collection(mongoComponent.database, collectionName, domainFormat, optRegistry)
 
-  Await.result(createIndexes(), 3.seconds)
-
-  def createIndexes(): Future[Seq[String]] =
-    if (indexes.nonEmpty) {
-      collection
-        .createIndexes(indexes)
-        .toFuture
-        .recover {
-          case throwable: Throwable =>
-            logger.error("Failed to create indexes", throwable)
-            Seq.empty
-        }
-    } else {
+  Await.result(ensureIndexes(), 5.seconds)
+
+  def ensureIndexes(): Future[Seq[String]] = {
+    if (indexes.isEmpty) {
       logger.info("Skipping Mongo index creation as no indexes supplied")
-      Future.successful(Seq.empty)
     }
+    val futureZippedIndexes =
```

Comment:
Would the following be simpler?
`Future.traverse(indexes){ index =>
  collection.createIndex(index)
   .recoverWith { case IndexConflict(e) if rebuildIndex => ...
  }
}`